compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -491,6 +500,35 @@ impl<'a> StringReader<'a> {
         FatalError.raise()
     }
 
+    // See RFC 3101.
+    fn report_reserved_prefix(&self, start: BytePos) {
```

Comment:
```suggestion
    fn maybe_report_unknown_prefix(&self, start: BytePos) {
```

Reply:
Why `maybe`? In all cases it reports it. Either as a lint, or as a hard error.

Reply:
Fair. I then I might like `report_unknown_prefix_as_lint_or_error`